### DISCUSSION TITLE
GSOC project discussions

### DISCUSSION THREAD

**Discussion by adkas0971** [koordinates/kart#575] [Google Soc / SoD]
Hello , My name is Amandeep Kumar. I have made projects on web scraping . I have seen project idea list and found
two projects interesting "Kart CLI Help Improvements" and "Attachments Support".

Can anyone in the community please guide me how to get started with both these projects?

@rcoup  , @craigds for "Kart CLI Help Improvements"

@olsen232  for ""Attachments Support"

**Comment by rcoup** [comment]
Hi @adkas0971,

Part of GSoC is writing your proposal: you will need to do some reading and research into how to go about implementing the project you select: while technically you could apply for two I wouldn't recommend it. This will mean working through how it will work and what steps will be required.

For CLI Help you might want to look at whether there are similar plugins for [Click](https://click.palletsprojects.com/) (the CLI framework we use); or how other cross-platform Python projects do it (eg: [aws-cli](https://github.com/aws/aws-cli/blob/develop/awscli/help.py)'s `help` implementation)

For Attachments I'll let @olsen232  chime in.

**Reply by olsen232** [reply]
I've written some more detail about attachments here: https://github.com/koordinates/kart/issues/583
It's not necessarily a very self explanatory project, as it involves a good deal of understanding of what Kart does now, in order to understand what it should do to support attachments. I hope this doesn't obviate the "read and research" requirement - but my plan is still at a very high level, there's lots more thinking to be done.
The [GSOC timeline](https://developers.google.com/open-source/gsoc/timeline) says that coding starts on June 13 so right now we're in "Potential GSoC contributors discuss application ideas with mentoring organizations" if I understand correctly - no need to jump in to coding right away.
